cluster-autoscaler/cloudprovider/clusterapi/clusterapi_utils.go

```diff
@@ -153,3 +161,36 @@ func normalizedProviderString(s string) normalizedProviderID {
 	split := strings.Split(s, "/")
 	return normalizedProviderID(split[len(split)-1])
 }
+
+func scaleFromZeroEnabled(annotations map[string]string) bool {
+	cpu := annotations[cpuKey]
+	mem := annotations[memoryKey]
+
+	if cpu != "" && mem != "" {
+		return true
+	}
+	return false
+}
+
+func parseKey(annotations map[string]string, key string) (resource.Quantity, error) {
+	if val, exists := annotations[key]; exists && val != "" {
+		return resource.ParseQuantity(val)
+	}
+	return zeroQuantity.DeepCopy(), nil
+}
+
+func parseCPUCapacity(annotations map[string]string) (resource.Quantity, error) {
+	return parseKey(annotations, cpuKey)
+}
+
+func parseMemoryCapacity(annotations map[string]string) (resource.Quantity, error) {
+	return parseKey(annotations, memoryKey)
```

Comment:
this is comparing not matching formats, i.e this would expect 
```
return resource.ParseQuantity(fmt.Sprintf("%vMi", val))
```

Reply:
do we need to handle the case where a user puts the unit type in their annotation?

Reply:
The user does not set this value, our controllers do. We set a contract that we pass a number representing Mb `machine.openshift.io/memoryMb` any additional check here wouldn't hurt though.